src/queue/index.js

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2018  Shivam Tripathi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+import * as Error from '../helpers/errors';
+import amqp from 'amqplib';
+import {isNotDefined} from '../helpers/utils';
+import log from '../helpers/logger';
+
+
+/**
+ * QUEUE_NAME - Queue name for importing data.
+ * @type {string} QUEUE_NAME
+ **/
+export const QUEUE_NAME = 'bookbrainz-import';
+
+/**
+ * Connection - Object containing functions to create and close RMQ connections.
+ * @type {Object} Connection
+ * @property {function} connect - Returns a connection object promise
+ * @property {function} shutdown - Takes in a connection promise and closes the
+ * 		connection
+ **/
+export const Connection = {
+	connect: function connect() {
+		let connection = null;
+		try {
+			connection = amqp.connect('amqp://localhost');
+		}
+		catch (err) {
+			Error.raiseError(Error.CONNECTION_ERROR)(err);
+		}
+		return connection;
+	},
+	shutdown: function shutdown(connectionPromise) {
+		if (isNotDefined(connectionPromise)) {
+			Error.undefinedValue('Connection.shutdown:: connectionPromise');
+			return null;
+		}
+
+		return connectionPromise
+			.then(connection => connection.close())
+			.catch(Error.raiseError(Error.CONNECTION_CLOSE_ERROR));
+	}
+};
+
+/** Class representing a RMQ Queue. */
+export class Queue {
+	/**
+     * Create a channel promise
+     * @param {Promise} connectionPromise - Connection promise
+     */
+	constructor(connectionPromise) {
+		if (isNotDefined(connectionPromise)) {
+			Error.undefinedValue('Queue.constructor:: connectionPromise');
+			return null;
+		}
+		this.channelPromise = connectionPromise
+			.then(conn => conn.createChannel())
+			.catch(Error.raiseError(Error.CHANNEL_ERROR));
+	}
+
+	/**
+     * Push a message into the queue
+     * @param {string} msg - The string containing message to be pushed into RMQ
+     */
+	push(msg) {
+		if (isNotDefined(msg)) {
+			log.error('Invalid message! Skipping.');
+			return;
+		}
+
+		const serializedMessage = JSON.stringify(msg);
```

Comment:
Definitely sure that you want to call JSON.stringify on a string? If so, please add a comment, since it's slightly unusual...